### PR TITLE
Make Rebalance() a little easier to read.

### PR DESCRIPTION
**Pull request by obnoxxx**
Signed-off-by: Michael Adam <obnox@redhat.com>

While studying the ring allocator code, I found Rebalance() a little
difficult to read/understand. I tried to make it more readable by
introducing a few helper variables etc.

What do others think - is this useful?

### CODE REVIEW

File: apps/glusterfs/allocator_simple_ring.go [hunk 1]
```diff
@@ -134,20 +134,26 @@ func (s *SimpleAllocatorRing) Rebalance() {
...
-		node := i % len(zones[zone])
+		nodes := zones[zone]
+		node := i % len(nodes)
+		devices := nodes[node]
 
 		// pop device
-		device, zones[zone][node] = zones[zone][node][len(zones[zone][node])-1], zones[zone][node][:len(zones[zone][node])-1]
+		device = devices[len(devices)-1]
+		devices = devices[:len(devices)-1]
+		nodes[node] = devices
```
Comment (lpabon) [inline apps/glusterfs/allocator_simple_ring.go]: That is _so_ much easier to read.

File: apps/glusterfs/allocator_simple_ring.go [hunk 2]
```diff
@@ -134,20 +134,26 @@ func (s *SimpleAllocatorRing) Rebalance() {
...
+
 		list = append(list, *device)
 
-		// delete node
-		if len(zones[zone][node]) == 0 {
-			zones[zone] = append(zones[zone][:node], zones[zone][node+1:]...)
+		if len(devices) == 0 {
+			// delete node
+			nodes = append(nodes[:node], nodes[node+1:]...)
```
Comment (lpabon) [inline apps/glusterfs/allocator_simple_ring.go]: nice

File: apps/glusterfs/allocator_simple_ring.go [hunk 3]
```diff
@@ -134,20 +134,26 @@ func (s *SimpleAllocatorRing) Rebalance() {
...
+			zones[zone] = nodes
+		}
 
+		if len(nodes) == 0 {
 			// delete zone
-			if len(zones[zone]) == 0 {
-				zones = append(zones[:zone], zones[zone+1:]...)
-			}
+			zones = append(zones[:zone], zones[zone+1:]...)
```
Comment (lpabon) [inline apps/glusterfs/allocator_simple_ring.go]: nice!

**Comment by MohamedAshiqrh** [comment]
@obnoxxx Nice work. Really helpful.

**Pull request merged**